src/test/java/org/elasticsearch/percolator/PercolatorTests.java

```diff
@@ -917,7 +917,85 @@ public void testPercolateMultipleIndicesAndAliases() throws Exception {
         for (PercolateResponse.Match match : response) {
             assertThat(match.getIndex().string(), equalTo("test2"));
         }
+    }
+
+    @Test
+    public void testPercolateWithAliasFilter() throws Exception {
+        assertAcked(prepareCreate("my-index")
+                        .addMapping(PercolatorService.TYPE_NAME, "a", "type=string,index=not_analyzed")
+                        .addAlias(new Alias("a").filter(FilterBuilders.termFilter("a", "a")))
+                        .addAlias(new Alias("b").filter(FilterBuilders.termFilter("a", "b")))
+                        .addAlias(new Alias("c").filter(FilterBuilders.termFilter("a", "c")))
+        );
+        client().prepareIndex("my-index", PercolatorService.TYPE_NAME, "1")
+                .setSource(jsonBuilder().startObject().field("query", matchAllQuery()).field("a", "a").endObject())
+                .get();
+        client().prepareIndex("my-index", PercolatorService.TYPE_NAME, "2")
+                .setSource(jsonBuilder().startObject().field("query", matchAllQuery()).field("a", "b").endObject())
+                .get();
+        refresh();
+
+        // Specifying no filter, sorting, aggs then a different code path is executed (not realtime, but near realtime),
+        // but because we work here with aliases we forcefully go into that mode
```

Comment:
The translog plays no part here. If no filter/agg/sorting is defined it will directly operate on the queries map in PercolatorQueryRegistry and no execute a Lucene search to fetch the queries to evaluate.